src/Tmds.Ssh/SshClientSettings.SshConfig.cs

```diff
@@ -0,0 +1,238 @@
+                break;
+
+            case SshConfig.StrictHostKeyChecking.AcceptNew:
+                settings.UpdateKnownHostsFileAfterAuthentication = true;
+                // Disallow changed. Allow unknown.
+                settings.HostAuthentication =
+                    (KnownHostResult knownHostResult, SshConnectionInfo connectionInfo, CancellationToken cancellationToken)
+                        => ValueTask.FromResult(knownHostResult == KnownHostResult.Unknown);
+                break;
+
+            case SshConfig.StrictHostKeyChecking.Ask:
+                settings.UpdateKnownHostsFileAfterAuthentication = true;
+                // Disallow changed, and ask for unknown keys.
+                if (options.HostAuthentication is HostAuthentication authentication)
+                {
+                    settings.HostAuthentication =
+                        (KnownHostResult knownHostResult, SshConnectionInfo connectionInfo, CancellationToken cancellationToken) =>
+                        {
+                            if (knownHostResult == KnownHostResult.Changed)
+                            {
+                                return ValueTask.FromResult(false);
+                            }
+                            return authentication(knownHostResult, connectionInfo, cancellationToken);
+                        };
+                }
+                else
+                {
+                    settings.HostAuthentication = delegate { return ValueTask.FromResult(false); };
+                }
+                break;
+
+            case SshConfig.StrictHostKeyChecking.Yes:
+            default:
+                settings.UpdateKnownHostsFileAfterAuthentication = false;
+                settings.HostAuthentication = delegate { return ValueTask.FromResult(false); };
+                break;
+        }
+
+        return settings;
+    }
+
+    private static IReadOnlyList<Credential> DetermineCredentials(SshConfig config)
+    {
+        bool addPubKeyCredentials = config.PubKeyAuthentication ?? true;
+        bool addGssApiCredentials = config.GssApiAuthentication ?? false;
+
+        ReadOnlySpan<Name[]> authPreferences = [
+            config.PreferredAuthentications ?? Array.Empty<Name>(),
+            DefaultPreferredAuthentications
+        ];
+
+        List<Credential> credentials = new();
+
+        foreach (var preferredAuthentications in authPreferences)
+        {
+            foreach (var algorithm in preferredAuthentications)
+            {
+                if (algorithm == AlgorithmNames.GssApiWithMic)
+                {
+                    if (addGssApiCredentials)
+                    {
+                        bool delegateCredential = config.GssApiDelegateCredentials ?? false;
+                        string? serverIdentity = config.GssApiServerIdentity;
+                        credentials.Add(new KerberosCredential(credential: null, delegateCredential, serverIdentity));
```

Comment:
Both OpenSSH and libssh seem to prefix the provided service identity with `host@`.

I assume we should do the same and I'll push a commit that adopts the same behavior.

This means the value passed to the `KerberosCredential` constructor must no longer include a `host/` prefix.

Let me know if you have any concerns.

Reply:
This is the change: https://github.com/tmds/Tmds.Ssh/pull/200/commits/81c5cf0a1b93a0a6de0ad7d343df5f1d9c79ce17.

Reply:
That's fine if you wish to restrict yourself to how OpenSSH works. I personally think the actual API interface can be fully flexible with accepting the full SPN in case someone actually wants that and you can restrict yourself in the config parser to how OpenSSH works.

Either all is ultimately fine I just personally don't see why it's necessary to restrict the actual API.

Reply:
I don't have a strong opinion, so I've reverted this and prefix with `host@` in the SshConfig.

I assume `host@` is preferable to `host/` since that is what OpenSSH/libssh are using.

I changed the name on the KerberosCredential to targetName so it matches with "NegotiateAuthentication.TargetName".

Reply:
> I assume host@ is preferable to host/ since that is what OpenSSH/libssh are using.

It needs to be `/` and not `@` to support SSPI on Windows. .NET automatically converts it to the `@` on the platforms it needs to be on for us.